components/user-mgt/org.wso2.carbon.identity.user.pre.update.profile.action/src/main/java/org/wso2/carbon/identity/user/pre/update/profile/action/internal/management/PreUpdateProfileActionDTOModelResolver.java

```diff
@@ -123,7 +133,8 @@ public ActionDTO resolveForUpdateOperation(ActionDTO updatingActionDTO, ActionDT
         Map<String, ActionPropertyForDAO> properties = new HashMap<>();
         // Action Properties updating operation is treated as a PUT in DAO layer. Therefore if no properties are updated
         // the existing properties should be sent to the DAO layer.
-        List<String> attributes = getResolvedUpdatingOrExistingAttributes(updatingActionDTO, existingActionDTO);
+        List<String> attributes = getResolvedUpdatingOrExistingAttributes(updatingActionDTO, existingActionDTO,
+                tenantDomain);
         if (!attributes.isEmpty()) {
             properties.put(ATTRIBUTES, createActionProperty(attributes));
         }
```

Comment:
Ideally it would be better if we can give this part to the `getResolvedUpdatingOrExistingAttributes()` method by passing the properties map to it. Then this method looks more cleaner.